# Genome Assembly {#assembly}

**Note: many of the instructions for this workshop are adapted from work by Phillip Compeau that can be found at https://compeau.cbd.cmu.edu/online-education/sars-cov-2-software-assignments/ as well as Galaxy tutorials found at https://training.galaxyproject.org/training-material/topics/assembly/tutorials/mrsa-nanopore/tutorial.html**

## Data

In this workshop you have produced genome sequence data in the lab.
This file is in FASTQ format, in which each read is represented over four lines described as follows.

- A header beginning with the “@” symbol and labeling the read. Notice that because these are read pairs, the first two reads’ headers are the same and end with “/1” and “/2”.
- The read as a sequence of nucleotides.
- A line containing a “+” symbol to indicate the end of nucleotides. This line may have more header information, but doesn’t for this dataset.
- A collection of ASCII symbols representing Phred quality scores. The i-th quality score corresponds to the i-th nucleotide in the read. Phred scores are helpful for a variety of reasons. For example, researchers will throw out reads having a significant number of bases that do not meet a certain Phred score threshold, especially if the reads have high coverage.

![](https://www.drive5.com/usearch/manual/fastq_fig.jpg)

## Getting started with Galaxy

To to all of our work, including genome assembly and annotation, we will use the 
service Galaxy, an open source project that allows us to run often testy bioinformatics software in the cloud without the hassle of dealing with local installations. 
Please follow these step-by-step instructions to register on Galaxy:

1. Create an account on Galaxy at https://usegalaxy.org/login. 
Your “public name” can be whatever you like, but you should fill out all fields. After creating an account, log in to Galaxy.

2. Click on the plus '+' on the top right of the page and create a new "history" (i.e., project that will contain data and the results of software runs) with a name that is relevant to the data you are analyzing.
Some of you are analyzing *E. coli* and others *H. para* so please name accordingly in case you want to try another analysis later.
All of the following analysis will be performed under this history. If you run a second analysis make sure to make a new history!

## Getting your sequencing data

1. Click on the 'Upload Data' button on the left side of the screen.
Ask the instructor about uploading the data from your sample onto Galaxy.

2. When the job turns green, click on the “eye” symbol next to one of your fastq files under your history on the right side of the page. The page may take a few seconds to load, and Galaxy will show you the first megabyte of your read file.

## Assembling a genome

Now that we understand our data a bit better, we are ready to assemble the genome from our data using Flye.

1. Under Genomic File Manipulation - Nanopore on the left side of the page, you will see various tools. 
Click on Flye, which will take you to its homepage.
2. You need to select the data that Flye will use to assemble your genome. Under Input reads, click on the Multiple Datasets button. 
3. Click the drop down arrow and select each dataset one at a time.

Leave all the remaining parameters as default.

Click Run!

The assembly will show up as green in your history when it has completed. This should take ~10-30 minutes depending on the size of the genome.

## Quality checking the assembly

Now that Flye has completed, we will analyze the results. 

Quast (Gurevich et al. 2013) is a tool providing quality metrics for assemblies, and can also be used to compare multiple assemblies. The tool can also take an optional reference file as input, and will provide complementary metrics. QUAST stands for QUality ASsessment Tool. 
1. From the left menus select Genomics Analysis - Assembly - Quast

2. Set parameters as
  - Assembly Mode: Individual
  - Select your assembly
  - Leave all other parameters as defaults

3. Run tool (it should be done in a few minutes)

4. View the HTML report from QUAST by clicking on the 'eye' in the analysis on the right. The Quast tool outputs assembly metrics as an html file with metrics and graphs. 
  - How many contigs (i.e. genome fragments) did your assembly produce? 
  - What is the total read length? Is this what you expect from your sample? (Check out the [reference E. coli](https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_000007445.1/) or the [reference H. parainfluenzae](https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_031191205.1/))
  - What is the [N50](https://en.wikipedia.org/wiki/N50,_L50,_and_related_statistics) value?
  
Congratulations! You have assemble a bacterial genome that no one else has ever seen before!