# Genome Alignment {#align}

## Obtaining a reference genome

To align your assembled genome against a reference genome, we first need to import this reference to Galaxy. The reference we will use for Escherichia coli is a uropathogenic (i.e UTI causing) strain [O6:K2:H1 CFT073 (UPEC)](https://www.genome.jp/kegg-bin/show_organism?org=ecc). 

1. Select the “NCBI Accession Download” tool under “Get Data” on the left side of the Galaxy page. 
2. Under “Select source for IDs”, select “Direct Entry”.
3. Under ID List, type NC_004431 (for E. coli) or NZ_CP133470.1	(for H. para).
4. Click "Run Tool"

## Aligning two samples 

Now we are ready to align the two genomes. 

1. Under "Multiple Alignments" select "Mummer"
2. For sequence 1, select the reference by clicking the "..." and selecting NC_004431.
3. For sequence 2, select the contigs that you produced with Flye. 
4. Keep the other parameters default.
5. Click "Run Tool"

This should take a couple of minutes.

6. In the righthand box labeled "Mummer"..."plot" click the "eye" to view the alignment.

Diagonal lines indicate sections of the sample genome that align with the reference as in the following example:

![](https://mummer4.github.io/tutorial/img/dotplot.gif)

Our E. coli genes often turn out a lot messier than this example.