# Genome Annotation {#annot}

## Annotation

Finally, we will annotate the genome, meaning that we will identify putative genes and then compare these genes against a database of known genes to find which ones align the best.

The identification of putative genes can be done in a variety of ways. One way to find these regions requires the observation that regions of DNA that are translated into protein start with a start codon (ATG) and end with a stop codon (TAG, TAA, or TGA). If a genome were random, then after encountering a start codon, we would expect to see a stop codon after about 64/3 ~ 21 codons. But real genes are typically much, much longer than 21 codons. As a result, in simple organisms like viruses and bacteria, a great way to find genes is to look for long stretches of codons connecting a start codon to a stop codon, with no intermediate stop codons.

Once we have found putative genes along the genome, we can compare them against a database of genes using an algorithm called BLAST.

These two steps are taken by our next tool, called Prokka, which is used to annotate the genomes of viruses, bacteria, and archaea.

1. Select Prokka under Genomics Analysis - Annotation on the left. 
2. Under “Contigs to annotate”, select your contigs from the Flye run. 
3. Set “Kingdom” to “Bacteria” and leave all other parameters to default. Note that Prokka only annotates contigs of length at least 200.
4. Click “Run”.

The entire process of annotating your genome should not take more than a few minutes.

Prokka produces a collection of output files in the history on the right. 
- The .fna file contains the contigs included in the annotation.
- The .gff file contains regions identified by Prokka as genes. 
- The .ffn file contains nucleotide sequences of the annotated genes, and the .faa file contains their translated amino acid sequences.

More detailed descriptions of the outputs can be found at https://github.com/tseemann/prokka#output-files

## Viewing the annotation

Prokka contains the annotation information in these files, but what a biologist would really like is a visualization of its genes and what they have been predicted to be. So to visualize our annotation contained in the .gff file, we will use a genome browser tool called “JBrowse”

1. Click (on the left side of screen) "Statistics and Visualization" > "Graph/Display data" > "Jbrowse"
2. Under “Reference genome to display” select “use genome from history”
3. Under “Select reference genome” select the .fna file from the prokka annotation outputs.
4. Click “insert track group” and then “insert annotation track”
5. Under “GFF/GFF3/BED Track Data” select the .gff file from the prokka annotation outputs
6. Click “Run tool”
7. JBrowse should be very quick. When it is finished, click view file, which is an HTML file that we can view in the browser. (It may take a moment to load.)
8 . On the left side of the page, click on “Prokka on data XXX: gff” to show our annotation of the genome. 
  - The arrows point in the direction to indicate which strand of the genome the gene is found on. 
  - Click on a gene to obtain information about its length and what in the database it was similar to (if anything). 