<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 Genome Assembly | Galaxy Bacterial Genomics Workshop</title>
  <meta name="description" content="Step-by-step guide to a bacterial genomics project in Galaxy." />
  <meta name="generator" content="bookdown 0.35 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 Genome Assembly | Galaxy Bacterial Genomics Workshop" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Step-by-step guide to a bacterial genomics project in Galaxy." />
  <meta name="github-repo" content="rachelss/Galaxy_genomes" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 Genome Assembly | Galaxy Bacterial Genomics Workshop" />
  
  <meta name="twitter:description" content="Step-by-step guide to a bacterial genomics project in Galaxy." />
  

<meta name="author" content="Rachel Schwartz" />


<meta name="date" content="2024-01-17" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="intro.html"/>
<link rel="next" href="align.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>



<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Galaxy Genomics Workshop</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About this workshop</a></li>
<li class="chapter" data-level="2" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>2</b> Introduction</a>
<ul>
<li class="chapter" data-level="2.1" data-path="intro.html"><a href="intro.html#introduction"><i class="fa fa-check"></i><b>2.1</b> Introduction</a></li>
<li class="chapter" data-level="2.2" data-path="intro.html"><a href="intro.html#case-studies"><i class="fa fa-check"></i><b>2.2</b> Case Studies</a></li>
<li class="chapter" data-level="2.3" data-path="intro.html"><a href="intro.html#course-objectives"><i class="fa fa-check"></i><b>2.3</b> Course Objectives</a></li>
<li class="chapter" data-level="2.4" data-path="intro.html"><a href="intro.html#genomic-data"><i class="fa fa-check"></i><b>2.4</b> Genomic data</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="assembly.html"><a href="assembly.html"><i class="fa fa-check"></i><b>3</b> Genome Assembly</a>
<ul>
<li class="chapter" data-level="3.1" data-path="assembly.html"><a href="assembly.html#data"><i class="fa fa-check"></i><b>3.1</b> Data</a></li>
<li class="chapter" data-level="3.2" data-path="assembly.html"><a href="assembly.html#getting-started-with-galaxy"><i class="fa fa-check"></i><b>3.2</b> Getting started with Galaxy</a></li>
<li class="chapter" data-level="3.3" data-path="assembly.html"><a href="assembly.html#getting-your-sequencing-data"><i class="fa fa-check"></i><b>3.3</b> Getting your sequencing data</a></li>
<li class="chapter" data-level="3.4" data-path="assembly.html"><a href="assembly.html#assembling-a-genome"><i class="fa fa-check"></i><b>3.4</b> Assembling a genome</a></li>
<li class="chapter" data-level="3.5" data-path="assembly.html"><a href="assembly.html#quality-checking-the-assembly"><i class="fa fa-check"></i><b>3.5</b> Quality checking the assembly</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="assembly.html"><a href="assembly.html#genome-metrics"><i class="fa fa-check"></i><b>3.5.1</b> Genome metrics</a></li>
<li class="chapter" data-level="3.5.2" data-path="assembly.html"><a href="assembly.html#assessing-genome-information"><i class="fa fa-check"></i><b>3.5.2</b> Assessing genome information</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="align.html"><a href="align.html"><i class="fa fa-check"></i><b>4</b> Genome Alignment</a>
<ul>
<li class="chapter" data-level="4.1" data-path="align.html"><a href="align.html#obtaining-a-reference-genome"><i class="fa fa-check"></i><b>4.1</b> Obtaining a reference genome</a></li>
<li class="chapter" data-level="4.2" data-path="align.html"><a href="align.html#aligning-two-samples"><i class="fa fa-check"></i><b>4.2</b> Aligning two samples</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="align2.html"><a href="align2.html"><i class="fa fa-check"></i><b>5</b> Align data to reference</a>
<ul>
<li class="chapter" data-level="5.1" data-path="align2.html"><a href="align2.html#align-1"><i class="fa fa-check"></i><b>5.1</b> Align</a></li>
<li class="chapter" data-level="5.2" data-path="align2.html"><a href="align2.html#merge"><i class="fa fa-check"></i><b>5.2</b> Merge</a></li>
<li class="chapter" data-level="5.3" data-path="align2.html"><a href="align2.html#view"><i class="fa fa-check"></i><b>5.3</b> View</a></li>
<li class="chapter" data-level="5.4" data-path="align2.html"><a href="align2.html#variant-calling"><i class="fa fa-check"></i><b>5.4</b> Variant calling</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="annot.html"><a href="annot.html"><i class="fa fa-check"></i><b>6</b> Genome Annotation</a>
<ul>
<li class="chapter" data-level="6.1" data-path="annot.html"><a href="annot.html#annotation"><i class="fa fa-check"></i><b>6.1</b> Annotation</a></li>
<li class="chapter" data-level="6.2" data-path="annot.html"><a href="annot.html#viewing-the-annotation"><i class="fa fa-check"></i><b>6.2</b> Viewing the annotation</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="gene.html"><a href="gene.html"><i class="fa fa-check"></i><b>7</b> Working with Target Genes</a>
<ul>
<li class="chapter" data-level="7.1" data-path="gene.html"><a href="gene.html#target-genes-in-with-a-personal-computer-if-web-only-skip-to-6.2"><i class="fa fa-check"></i><b>7.1</b> Target genes in with a personal computer (if web-only skip to 6.2)</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="gene.html"><a href="gene.html#download-and-install-jbrowse"><i class="fa fa-check"></i><b>7.1.1</b> Download and install Jbrowse</a></li>
<li class="chapter" data-level="7.1.2" data-path="gene.html"><a href="gene.html#obtaining-the-data-from-galaxy"><i class="fa fa-check"></i><b>7.1.2</b> Obtaining the data from Galaxy</a></li>
<li class="chapter" data-level="7.1.3" data-path="gene.html"><a href="gene.html#your-genome-in-jbrowse-desktop"><i class="fa fa-check"></i><b>7.1.3</b> Your genome in Jbrowse Desktop</a></li>
<li class="chapter" data-level="7.1.4" data-path="gene.html"><a href="gene.html#annotation-in-jbrowse-desktop"><i class="fa fa-check"></i><b>7.1.4</b> Annotation in Jbrowse Desktop</a></li>
<li class="chapter" data-level="7.1.5" data-path="gene.html"><a href="gene.html#finding-genes-in-jbrowse-desktop"><i class="fa fa-check"></i><b>7.1.5</b> Finding genes in Jbrowse desktop</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="gene.html"><a href="gene.html#target-genes-in-galaxy-skip-this-if-using-jbrowse-on-your-computer"><i class="fa fa-check"></i><b>7.2</b> Target genes in Galaxy (skip this if using JBrowse on your computer)</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="gene.html"><a href="gene.html#finding-gene-locations"><i class="fa fa-check"></i><b>7.2.1</b> Finding gene locations</a></li>
<li class="chapter" data-level="7.2.2" data-path="gene.html"><a href="gene.html#observing-genes-in-the-genome"><i class="fa fa-check"></i><b>7.2.2</b> Observing genes in the genome</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="gene.html"><a href="gene.html#comparing-genes-between-strains-optional"><i class="fa fa-check"></i><b>7.3</b> Comparing genes between strains (optional)</a></li>
<li class="chapter" data-level="7.4" data-path="gene.html"><a href="gene.html#comparing-genes-to-a-database"><i class="fa fa-check"></i><b>7.4</b> Comparing genes to a database</a></li>
<li class="chapter" data-level="7.5" data-path="gene.html"><a href="gene.html#comparing-across-strains"><i class="fa fa-check"></i><b>7.5</b> Comparing across strains</a></li>
<li class="chapter" data-level="7.6" data-path="gene.html"><a href="gene.html#more-genes"><i class="fa fa-check"></i><b>7.6</b> More genes</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Galaxy Bacterial Genomics Workshop</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="assembly" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">Chapter 3</span> Genome Assembly<a href="assembly.html#assembly" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><strong>Note: many of the instructions for this workshop are adapted from work by Phillip Compeau that can be found at <a href="https://compeau.cbd.cmu.edu/online-education/sars-cov-2-software-assignments/" class="uri">https://compeau.cbd.cmu.edu/online-education/sars-cov-2-software-assignments/</a> as well as Galaxy tutorials found at <a href="https://training.galaxyproject.org/training-material/topics/assembly/tutorials/mrsa-nanopore/tutorial.html" class="uri">https://training.galaxyproject.org/training-material/topics/assembly/tutorials/mrsa-nanopore/tutorial.html</a></strong></p>
<div id="data" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> Data<a href="assembly.html#data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In this workshop you have produced genome sequence data in the lab.
This file is in FASTQ format, in which each read is represented over four lines described as follows.</p>
<ul>
<li>A header beginning with the “@” symbol and labeling the read. Notice that because these are read pairs, the first two reads’ headers are the same and end with “/1” and “/2”.</li>
<li>The read as a sequence of nucleotides.</li>
<li>A line containing a “+” symbol to indicate the end of nucleotides. This line may have more header information, but doesn’t for this dataset.</li>
<li>A collection of ASCII symbols representing Phred quality scores. The i-th quality score corresponds to the i-th nucleotide in the read. Phred scores are helpful for a variety of reasons. For example, researchers will throw out reads having a significant number of bases that do not meet a certain Phred score threshold, especially if the reads have high coverage.</li>
</ul>
<p><img src="https://www.drive5.com/usearch/manual/fastq_fig.jpg" /></p>
</div>
<div id="getting-started-with-galaxy" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> Getting started with Galaxy<a href="assembly.html#getting-started-with-galaxy" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To to all of our work, including genome assembly and annotation, we will use the
service Galaxy, an open source project that allows us to run bioinformatics software in the cloud without the hassle of dealing with local installations. You should <a href="https://galaxyproject.org/galaxy-project/">read more about the Galaxy</a> service, which is funded by the National Institute of Health and the National Science Foundation.
Please follow these step-by-step instructions to register on Galaxy:</p>
<ol style="list-style-type: decimal">
<li><p>Create an account on Galaxy at <a href="https://usegalaxy.org/login" class="uri">https://usegalaxy.org/login</a>.
Your “public name” can be whatever you like, but you should fill out all fields. After creating an account, log in to Galaxy. <strong>Note that if at some point you have difficulties with this server, try <a href="https://usegalaxy.org.au/" class="uri">https://usegalaxy.org.au/</a></strong></p></li>
<li><p>Check your email and click the Galaxy link.</p></li>
<li><p>Click on the plus ‘+’ on the top right of the page and create a new “history” (i.e., project that will contain data and the results of software runs) with a name that is relevant to the data you are analyzing.
Some of you are analyzing <em>E. coli</em> and others <em>H. para</em> so please name accordingly in case you want to try another analysis later.
All of the following analysis will be performed under this history. If you run a second analysis make sure to make a new history!</p></li>
</ol>
</div>
<div id="getting-your-sequencing-data" class="section level2 hasAnchor" number="3.3">
<h2><span class="header-section-number">3.3</span> Getting your sequencing data<a href="assembly.html#getting-your-sequencing-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ol style="list-style-type: decimal">
<li><p>Download your data from the Google Drive that your instructor shared with you. You should see files ending in .fastq.gz . If you have a zipped folder of files make sure to unzip it. Do not unzip the fastq files.</p></li>
<li><p>In Galaxy, click on the ‘Upload Data’ button on the left side of the screen.</p></li>
<li><p>Click ‘Choose Local File’</p></li>
<li><p>Select all your .fastq.gz files and click Open</p></li>
<li><p>Click Start</p></li>
<li><p>When the jobs have completed and are all green on the right side of your main Galaxy page, click on the “eye” symbol next to one of your fastq files under your history on the right side of the page. The page may take a few seconds to load, and Galaxy will show you the first megabyte of your read file.</p></li>
</ol>
</div>
<div id="assembling-a-genome" class="section level2 hasAnchor" number="3.4">
<h2><span class="header-section-number">3.4</span> Assembling a genome<a href="assembly.html#assembling-a-genome" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Now that we understand our data a bit better, we are ready to assemble the genome from our data using Flye.</p>
<ol style="list-style-type: decimal">
<li>Under Genomic File Manipulation - Nanopore on the left side of the page, you will see various tools.
Click on Flye, which will take you to its homepage.</li>
<li>You need to select the data that Flye will use to assemble your genome. Under Input reads, click on the Multiple Datasets button.</li>
<li>Click the drop down arrow and select each dataset one at a time.</li>
</ol>
<p>Leave all the remaining parameters as default.</p>
<p>Click Run!</p>
<p>The assembly will show up as green in your history when it has completed. This should take ~10-30 minutes depending on the size of the genome.</p>
</div>
<div id="quality-checking-the-assembly" class="section level2 hasAnchor" number="3.5">
<h2><span class="header-section-number">3.5</span> Quality checking the assembly<a href="assembly.html#quality-checking-the-assembly" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Now that Flye has completed, we will analyze the results.</p>
<div id="genome-metrics" class="section level3 hasAnchor" number="3.5.1">
<h3><span class="header-section-number">3.5.1</span> Genome metrics<a href="assembly.html#genome-metrics" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Quast (Gurevich et al. 2013) is a tool providing quality metrics for assemblies, and can also be used to compare multiple assemblies. The tool can also take an optional reference file as input, and will provide complementary metrics. QUAST stands for QUality ASsessment Tool.</p>
<ol style="list-style-type: decimal">
<li><p>From the left menus select Genomics Analysis - Assembly - Quast</p></li>
<li><p>Set parameters as</p></li>
</ol>
<ul>
<li>Assembly Mode: Individual</li>
<li>Select your assembly</li>
<li>Leave all other parameters as defaults</li>
</ul>
<ol start="3" style="list-style-type: decimal">
<li><p>Run tool (it should be done in a few minutes)</p></li>
<li><p>View the HTML report from QUAST by clicking on the ‘eye’ in the analysis on the right. The Quast tool outputs assembly metrics as an html file with metrics and graphs.</p></li>
</ol>
<ul>
<li>How many contigs (i.e. genome fragments) did your assembly produce?</li>
<li>What is the total read length? Is this what you expect from your sample? (Check out the <a href="https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_000007445.1/">reference E. coli</a> or the <a href="https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_031191205.1/">reference H. parainfluenzae</a>)</li>
<li>What is the <a href="https://en.wikipedia.org/wiki/N50,_L50,_and_related_statistics">N50</a> value?</li>
</ul>
</div>
<div id="assessing-genome-information" class="section level3 hasAnchor" number="3.5.2">
<h3><span class="header-section-number">3.5.2</span> Assessing genome information<a href="assembly.html#assessing-genome-information" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Busco is a tool used to determine whether your genome has the expected genes for a bacterial genome.</p>
<ol style="list-style-type: decimal">
<li><p>From the left menus select Annotation - Busco</p></li>
<li><p>Run tool</p></li>
<li><p>Even if it fails click on the “eye” to view the summary analysis. You should see some information about your assembly, particularly how many of the expected genes were not found.</p></li>
</ol>
<ul>
<li>How many genes were missing?</li>
</ul>
<p>Congratulations! You have assemble a bacterial genome that no one else has ever seen before!</p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="intro.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="align.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/02-galaxy.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
